modules/secrets-store-aws.adoc

```diff
@@ -0,0 +1,370 @@
+          resources:
+            requests:
+              cpu: 50m
+              memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: "/etc/kubernetes/secrets-store-csi-providers"
+              name: providervol
+            - name: mountpoint-dir
+              mountPath: /var/lib/kubelet/pods
+              mountPropagation: HostToContainer
+      tolerations:
+      - operator: Exists
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/etc/kubernetes/secrets-store-csi-providers"
+        - name: mountpoint-dir
+          hostPath:
+            path: /var/lib/kubelet/pods
+            type: DirectoryOrCreate
+      nodeSelector:
+        kubernetes.io/os: linux
+----
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider[AWS documentation] so that it works properly with {product-title}.
+====
+
+.. Grant privileged access to the `csi-secrets-store-provider-aws` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z csi-secrets-store-provider-aws -n openshift-cluster-csi-drivers
+----
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f aws-provider.yaml
+----
+
+. Grant permission to allow the service account to read the AWS secret object:
+
+.. Create a directory to contain the credentials request by running the following command:
++
+[source,terminal]
+----
+$ mkdir credentialsrequest-dir-aws
+----
+
+.. Create a YAML file with the following configuration for the credentials request:
++
+.Example `credentialsrequest.yaml` file
+[source,yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: aws-provider-test
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+ifdef::aws-secrets-manager[]
+    statementEntries:
+    - action:
+      - "secretsmanager:GetSecretValue"
+      - "secretsmanager:DescribeSecret"
+      effect: Allow
+      resource: "arn:*:secretsmanager:*:*:secret:testSecret-??????"
+endif::aws-secrets-manager[]
+ifdef::aws-systems-manager-parameter-store[]
+    statementEntries:
+    - action:
+      - "ssm:GetParameter"
+      - "ssm:GetParameters"
+      effect: Allow
+      resource: "arn:*:ssm:*:*:parameter/testParameter*"
+endif::aws-systems-manager-parameter-store[]
+  secretRef:
+    name: aws-creds
+    namespace: my-namespace
+  serviceAccountNames:
+  - aws-provider
+----
+
+.. Retrieve the OIDC provider by running the following command:
++
+[source,terminal]
+----
+$ oc get --raw=/.well-known/openid-configuration | jq -r '.issuer'
```

Comment:
Could `jsonpath` be used instead of `jq` here?

Reply:
Unfortunately no - I asked about it and looked into it. But the command produces a raw text that has to be parsed, so jsonpath can't be used, and jq is really the way to do it here.